### PR TITLE
Fix un-correlate scalar subquery bug

### PR DESCRIPTION
**Pull request by Seaven**
```
SELECT * from t1   WHERE 1=1   and case when (SELECT v1 from t1 WHERE false) =2   then false   else true   end;
```

except result: 
```
+------+------+------+
| v1   | v2   | v3   |
+------+------+------+
|    2 |   20 |  200 |
|    1 |   10 |  100 |
+------+------+------+
2 rows in set (0.03 sec)
```

but
```
+------+------+------+
| v1   | v2   | v3   |
+------+------+------+
+------+------+------+
2 rows in set (0.03 sec)
```

Fix:
If assertOneRowNode input always empty, least fill one null row

### CODE REVIEW

File: be/src/exec/pipeline/assert_num_rows_operator.cpp [hunk 1]
```diff
@@ -18,7 +29,27 @@ Status AssertNumRowsOperator::close(RuntimeState* state) {
...
+bool AssertNumRowsOperator::has_output() const {
+    return _input_finished && _cur_chunk;
+}
+
+bool AssertNumRowsOperator::need_input() const {
+    return !_input_finished;
+}
+
+Status AssertNumRowsOperator::push_chunk(RuntimeState* state, const vectorized::ChunkPtr& chunk) {
+    if (!chunk) {
```
Comment (kangkaisen) [inline be/src/exec/pipeline/assert_num_rows_operator.cpp:44]: Needn't

File: be/src/exec/pipeline/assert_num_rows_operator.cpp [hunk 2]
```diff
@@ -18,7 +29,27 @@ Status AssertNumRowsOperator::close(RuntimeState* state) {
...
+        return Status::OK();
+    }
+
+    if (chunk->num_rows() <= 0) {
```
Comment (kangkaisen) [inline be/src/exec/pipeline/assert_num_rows_operator.cpp:48]: needn't

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneAssertOneRowRule.java [hunk 3]
```diff
@@ -28,13 +27,6 @@ public boolean check(OptExpression input, OptimizerContext context) {
             return lao.getGroupingKeys().isEmpty();
         }
 
-        // if child is limit 1, remove AssertOneRow node
-        if (OperatorType.LOGICAL_LIMIT.equals(input.getInputs().get(0).getOp().getOpType())) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PruneAssertOneRowRule.java:32]: Need UT cover

**Review by kangkaisen — DISMISSED** [review]

**Review by Pslydhh — DISMISSED** [review]

**Review by kangkaisen — APPROVED** [review]

File: be/src/exec/pipeline/assert_num_rows_operator.cpp [hunk 4]
```diff
@@ -32,28 +55,17 @@ StatusOr<vectorized::ChunkPtr> AssertNumRowsOperator::pull_chunk(RuntimeState* s
...
 }
 
 void AssertNumRowsOperator::set_finishing(RuntimeState* state) {
-    _is_finished = true;
+    _input_finished = true;
 }
 
 bool AssertNumRowsOperator::is_finished() const {
-    return _is_finished && !_cur_chunk;
+    return !_cur_chunk;
```
Comment (satanson) [inline be/src/exec/pipeline/assert_num_rows_operator.cpp:68]: for the first time is_finished() is invoked, this method would return true.

**Review by satanson — APPROVED** [review]

**Pull request merged**